### PR TITLE
Error if a hardware name is duplicated

### PR DESCRIPTION
**Pull request by AndyZe**
I fell into this pitfall when trying to launch a dual-arm system. The robot xacro was duplicated so hardware loading failed.

### CODE REVIEW

File: hardware_interface/src/resource_manager.cpp [hunk 1]
```diff
@@ -426,6 +434,7 @@ class ResourceStorage
     for (auto & interface : interfaces)
     {
       auto key = interface.get_full_name();
+      RCUTILS_LOG_INFO_NAMED("resource_manager", "Command interface: '%s'", key.c_str());
```
Comment (bijoua29) [inline hardware_interface/src/resource_manager.cpp:437]: Why are we logging command interfaces and not state interfaces above?

Comment (AndyZe) [reply hardware_interface/src/resource_manager.cpp]: oops, that was just some debug code. Deleting it.

**Review by bijoua29 — APPROVED** [review]

**Review by aprotyas — APPROVED** [review]

**Review by destogl — COMMENTED** [review]
I am uncertain if we want to throw an exception here. I'm aware that we are throwing some on different places, but I am not convinced anymore that this is the appropriate strategy.

Reasoning: if we have hardware loaded later in the process (there is no interface immediately, but with extending CM definitely doable) this would cause crash of everything. Not fun.

Other than that, this check should be done much earlier, this is fairly late in the loading process.

Reasoning: when we support later loading of hardware, we should check if the name is duplicated before loading.


And tests are missing for this.

**Comment by AndyZe** [comment]
@destogl can you save me some trouble and point me toward the line # where you want the check to be moved?

Sorry, but I won't have time to write a test.

**Review by bmagyar — APPROVED** [review]
I created a follow-up issue to refactor this, the functionality however is valuable and I think we should have it sooner than the ideal solution surfaces.

**Comment by bmagyar** [comment]
https://github.com/ros-controls/ros2_control/issues/705

**Pull request merged**